### PR TITLE
Moving Bastion Stack to workload template

### PR DESCRIPTION
**Pull request by sourabh3b**
*Issue #, if available:* : https://github.com/aws-quickstart/quickstart-aerospike/issues/33

*Description of changes:* : adding bastion stack in workload template

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

Images : 

**SSH to bastion host and aerospike verification**
Screen Shot 2021-09-01 at 4 45 30 PM

**Bastion instance is present in public subnet**
Screen Shot 2021-09-01 at 4 45 57 PM

**Aerospike instance is present in private subnet**
Screen Shot 2021-09-01 at 4 46 06 PM

### CODE REVIEW

File: templates/aerospike-cluster-entrypoint-new-vpc.template.yaml [hunk 1]
```diff
@@ -76,12 +70,6 @@ Metadata:
         default: "Namespace file"
       EBS:
         default: "EBS volume size"
-      BastionAMIOS:
-        default: Bastion AMI operating system
-      BastionInstanceType:
-        default: Bastion instance type
-      NumBastionHosts:
-        default: Number of bastion hosts
```
Comment (gargana) [inline templates/aerospike-cluster-entrypoint-new-vpc.template.yaml:84]: Should we not still let the customer choose how they want to configure these?

File: templates/aerospike-cluster-entrypoint-new-vpc.template.yaml [hunk 2]
```diff
@@ -115,42 +103,6 @@ Parameters:
...
-      - '4'
-    Default: '1'
-    Description: Number of bastion hosts to create.
-    Type: String
-  BastionAMIOS:
-    AllowedValues:
-      - Amazon-Linux2-HVM
-    Default: Amazon-Linux2-HVM
-    Description: Linux distribution for the AMI to be used for the bastion instances.
-    Type: String
```
Comment (gargana) [inline templates/aerospike-cluster-entrypoint-new-vpc.template.yaml:153]: If this is a single allowed value we should then probably not keep it as a parameter?

File: templates/aerospike-cluster-entrypoint-new-vpc.template.yaml [hunk 3]
```diff
@@ -442,11 +351,15 @@ Resources:
...
+        PublicSubnet2ID: !GetAtt VPCStack.Outputs.PublicSubnet2ID
+        PublicSubnet3ID: !If [ 3AZDeployment, !GetAtt VPCStack.Outputs.PublicSubnet3ID, !Ref "AWS::NoValue" ]
+        PrivateSubnet1ID: !GetAtt VPCStack.Outputs.PrivateSubnet1AID
+        PrivateSubnet2ID: !GetAtt VPCStack.Outputs.PrivateSubnet2AID
+        PrivateSubnet3ID: !If [ 3AZDeployment, !GetAtt VPCStack.Outputs.PrivateSubnet3AID, !Ref "AWS::NoValue" ]
 
 Outputs:
-  BastionHostIP:
-    Description: IP for bastion host.
-    Value: !GetAtt "BastionStack.Outputs.EIP1"
```
Comment (gargana) [inline templates/aerospike-cluster-entrypoint-new-vpc.template.yaml:449]: This we should probably surface in the Workload stack and then still surface here.

File: templates/aerospike-cluster-entrypoint-new-vpc.template.yaml [hunk 4]
```diff
@@ -115,42 +103,6 @@ Parameters:
...
-      - m4.4xlarge
-  NumBastionHosts:
-    AllowedValues:
-      - '1'
-      - '2'
-      - '3'
-      - '4'
-    Default: '1'
-    Description: Number of bastion hosts to create.
-    Type: String
```
Comment (gargana) [inline templates/aerospike-cluster-entrypoint-new-vpc.template.yaml:147]: We probably want to still take these but then pass them into the child stack.

File: templates/aerospike-cluster-entrypoint-new-vpc.template.yaml (hunk 2 above)
Comment (gargana) [reply templates/aerospike-cluster-entrypoint-new-vpc.template.yaml:153]: The same would happen then in the workload stack.

**Review by gargana — CHANGES_REQUESTED** [review]
Please can we address the concerns raised? Thanks

File: templates/aerospike-cluster-entrypoint-new-vpc.template.yaml (hunk 1 above)
Comment (sourabh3b) [reply templates/aerospike-cluster-entrypoint-new-vpc.template.yaml]: done

File: templates/aerospike-cluster-entrypoint-new-vpc.template.yaml (hunk 2 above)
Comment (sourabh3b) [reply templates/aerospike-cluster-entrypoint-new-vpc.template.yaml]: done

File: templates/aerospike-cluster-entrypoint-new-vpc.template.yaml (hunk 3 above)
Comment (sourabh3b) [reply templates/aerospike-cluster-entrypoint-new-vpc.template.yaml:449]: I added this to workload stack, so we would get EIP all the time from both the stacks

File: templates/aerospike-cluster-entrypoint-new-vpc.template.yaml (hunk 4 above)
Comment (sourabh3b) [reply templates/aerospike-cluster-entrypoint-new-vpc.template.yaml]: done

**Comment by sourabh3b** [comment]
@gargana please take a look, I resolved all comments. thanks for the review

**Review by gargana — APPROVED** [review]
LGTM - Shipping

**Pull request merged**